api/v1alpha1/ionoscloudloadbalancer_types.go

```diff
@@ -66,22 +46,45 @@ type IonosCloudLoadBalancerSpec struct {
 	//+kubebuilder:validation:XValidation:rule="self.port == oldSelf.port || oldSelf.port == 0",message="control plane endpoint port cannot be updated"
 	LoadBalancerEndpoint clusterv1.APIEndpoint `json:"loadBalancerEndpoint,omitempty"`
 
-	// Type is the type of load balancer to create.
-	// defaults to HA.
-	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="load balancer type cannot be updated"
-	//+kubebuilder:validation:Enum=HA;NLB;External
-	//+kubebuilder:default=HA
-	//+required
-	Type LoadBalancerType `json:"type"`
+	// LoadBalancerSource is the actual load balancer definition.
+	LoadBalancerSource `json:",inline"`
+}
+
+// LoadBalancerSource defines the source of the load balancer.
+type LoadBalancerSource struct {
+	// NLB is used for setting up a network load balancer.
+	//+optional
+	NLB *NLBSpec `json:"nlb,omitempty"`
 
+	// KubeVIP is used for setting up a highly available control plane.
+	//+optional
+	KubeVIP *KubeVIPSpec `json:"kubeVIP,omitempty"`
+
+	// External is used for making use of an external load balancer.
+	//+optional
+	External *ExternalLoadBalancerSpec `json:"external,omitempty"`
+}
+
+// NLBSpec defines the spec for a network load balancer.
+type NLBSpec struct {
 	// DatacenterID is the ID of the datacenter where the load balancer should be created.
 	// This field is required for NLB load balancers and needs to match the datacenter ID
```

Comment:
I think we can drop the "is required for NLB" part

Reply:
:+1: